src/base_alloc/base_alloc_linux.c

```diff
@@ -19,8 +19,13 @@ static UTIL_ONCE_FLAG Page_size_is_initialized = UTIL_ONCE_FLAG_INIT;
 static size_t Page_size;
 
 void *ba_os_alloc(size_t size) {
-    return mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS,
-                -1, 0);
+    void *ptr = mmap(NULL, size, PROT_READ | PROT_WRITE,
+                     MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    // this should be unnecessary but pairs of mmap/munmap do not reset
+    // asan's user-poisoning flags, leading to invalid error reports
+    // Bug 81619: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81619
+    utils_annotate_memory_defined(ptr, size);
```

Comment:
In this case, shouldn't this memory be marked with `utils_annotate_memory_inaccessible` in `ba_os_free` and `os_munmap`?

Reply:
In other places where we call ba_os_free, we mark the memory with utils_annotate_memory_inaccessible. The annotations are included in this function due to the known bug in ASan's user-poisoning flags, which does not reset properly with pairs of mmap/munmap. @lplewa, what do you think?

Reply:
We can add them - it's look like there is problem with an asan marking correctly pages on mmap/unmap so we can do it on our side.

Reply:
Done

Reply:
Due to ASAN errors, I've asked @szadam to revert those changes.

Reply:
Okey. https://github.com/oneapi-src/unified-memory-framework/actions/runs/10039436558/job/27744736615?pr=610